dune_client/client.py

```diff
@@ -126,9 +140,37 @@ def refresh(self, query: Query, ping_frequency: int = 5) -> ResultsResponse:
             )
             time.sleep(ping_frequency)
             status = self.get_status(job_id)
-
-        full_response = self.get_result(job_id)
         if status.state == ExecutionState.FAILED:
             self.logger.error(status)
             raise QueryFailed(f"{status}. Perhaps your query took too long to run!")
-        return full_response
+
+        return job_id
+
+    def refresh(self, query: Query, ping_frequency: int = 5) -> ResultsResponse:
+        """
+        Executes a Dune `query`, waits until execution completes,
+        fetches and returns the results.
+        Sleeps `ping_frequency` seconds between each status request.
+        """
+        job_id = self._refresh(query, ping_frequency=ping_frequency)
+        return self.get_result(job_id)
+
+    def refresh_csv(self, query: Query, ping_frequency: int = 5) -> ExecutionResultCSV:
+        """
+        Executes a Dune query, waits till execution completes,
+        fetches and the results in CSV format
+        (use it load the data directly in pandas.from_csv() or similar frameworks)
+        """
+        job_id = self._refresh(query, ping_frequency=ping_frequency)
+        return self.get_result_csv(job_id)
+
+    def refresh_into_dataframe(self, query: Query) -> Any:
+        """
+        Execute a Dune Query, waits till execution completes,
+        fetched and returns the result as a Pandas DataFrame
+
+        This is a convenience method that uses refresh_csv underneath
+        it assumes the caller has already called `import pandas`
+        data = self.refresh_csv(query).data
+        return pandas.read_csv(data)
```

Comment:
This could be implemented without increase the required dependencies of `dune-client`.
```
within this method, we could do:
```
def refresh_into_dataframe():
   try:
     import pandas 
   except ImportError:
     raise ...
   data = self.refresh_csv(query).data
   return pandas.read_csv(data)
```
This way we don't do top level import, that would increase overheads.
This would not pass the current linting policy AFAIK, we would need to few linting exceptions for this to be tolerated.

We can also simply not include this at all, but leave it in the documentation

Reply:
This is an excellent suggestion and I like it very much. Would totally be willing to include some lint exceptions in order to make this pass!